### PR TITLE
Product variation availiability joan

### PR DESCRIPTION
**Pull request by juanlopez4691**
Related to task https://app.asana.com/0/30743925268131/192230416969568/f

This includes a refactoring of previous code and a fix to the availability evaluation for products not present in any showroom, attending at its type (simple or variation).

Also, phpunit test code has been corrected to reflect those mentioned changes.

### CODE REVIEW

File: src/Plugin.php [hunk 1]
```diff
@@ -92,59 +93,67 @@ public static function woocommerce_available_variation(array $args, \WC_Product
...
-   * @param string $supply
-   *   Origin store of the product.
-   * @param int $stock_store
-   *   Amount of product units in the store.
-   * @param int $stock_showroom
-   *   Amount of product units in the showroom.
-   * @param string $delivery_time
-   *   Time needed to deliver the product.
+   * @param array $args
+   *   Array containing the necessary params.
```
Comment (juliquiron) [inline src/Plugin.php]: great, but you already said that it's an Array! I would rephrase it.

File: src/Plugin.php [hunk 2]
```diff
@@ -156,45 +165,53 @@ public static function getProductAvailability(string $supply, int $stock_store,
...
-   * @param string $supply
-   *   Store where the product is located.
-   * @param int $stock_store
-   *   Amount of product units in the store.
-   * @param int $stock_showroom
-   *   Amount of product units in the showroom.
-   * @param string $available_stores
-   *   Stores where the product is available.
+   * @param array $args
+   *   Array containing the necessary params.
```
Comment (juliquiron) [inline src/Plugin.php]: Same

**Comment by juanlopez4691** [comment]
@juliquiron Pushed a fixup with your suggested corrections. Also added string localisation and corrected unit tests code considering this. Thanks for your review!

**Comment by juanlopez4691** [comment]
>Sorry, I don't know what happened here, but we have a pretty big architectural design flaw, as the code in this repository won't ever be used on any other website than WOHN.

See yout comment in the related task, also supported by @stephansperling 
 https://app.asana.com/0/30743925268131/192230416969568)

>Joan López In light of follow-up tasks like Variable products: show availability in different stores, can we bundle this whole functionality of multiple stocks/warehouses in a dedicated separate plugin? (using hardcoded warehouses for now, UI configuration for that can follow later)

@sun Now, followed your suggestion about changing architecture approach:
>The alternative is to rename this plugin into custom-stocks and commit it (back) into the WOHN repo itself.
Started a PR for this: https://github.com/netzstrategen/wohn/compare/fix/custom-stock-availability-192230416969568-joan?expand=1

**Comment by juanlopez4691** [comment]
@sun A new PR is open in WOHN integrating this plugin as "custom-stock" plugin, following your reaquirements. Thanks.

See https://github.com/netzstrategen/wohn/pull/13

**Comment by sun** [comment]
@juanlopez4691 Please delete this repo after fully migrating & restoring all changes into the new PR.
